src/main/java/net/minecraftforge/common/world/WorldGenValidator.java

```diff
@@ -0,0 +1,122 @@
+
+        // Checks every biome for unregistered or broken worldgen elements.
+        for (Biome biome : biomeRegistry) {
+            BiomeGenerationSettings biomeGenerationSettings = biome.getGenerationSettings();
+            ResourceLocation biomeName = biome.getRegistryName();
+
+            biomeGenerationSettings.features().forEach(generationStageFeatureList ->
+                    generationStageFeatureList.forEach(configuredFeatureSupplier -> {
+                        ConfiguredFeature<?, ?> feature = configuredFeatureSupplier.get();
+                        if (!checkedElements.contains(feature)) {
+                            validate(feature, configuredFeatureRegistry, "ConfiguredFeature", ConfiguredFeature.DIRECT_CODEC, biomeName);
+                            checkedElements.add(feature);
+                        }
+                    }));
+
+            for (GenerationStep.Carving carvingStage : GenerationStep.Carving.values()) {
+                biomeGenerationSettings.getCarvers(carvingStage).forEach(configuredWorldCarverSupplier -> {
+                    ConfiguredWorldCarver<?> carver = configuredWorldCarverSupplier.get();
+                    if (!checkedElements.contains(carver)) {
+                        validate(carver, configuredCarverRegistry, "ConfiguredCarver", ConfiguredWorldCarver.DIRECT_CODEC, biomeName);
+                        checkedElements.add(carver);
+                    }
+                });
+            }
+
+            biomeGenerationSettings.structures().forEach(structureFeatureSupplier -> {
+                ConfiguredStructureFeature<?, ?> structureFeature = structureFeatureSupplier.get();
+                if (!checkedElements.contains(structureFeature)) {
+                    validate(structureFeature, configuredStructureRegistry, "ConfiguredStructure", ConfiguredStructureFeature.DIRECT_CODEC, biomeName);
+                    checkedElements.add(structureFeature);
+                }
+            });
+        }
+    }
+
+    /**
+     * Handles checking if the worldgen element is broken (unable to be turned into JSON) or unregistered.
+     * If an issue is found, this will pause the IDE so the modder can come here and see the error and the worldgenElement for which element is the issue.
+     * The console output will have the Forge message as last thing when paused too.
+     */
+    private static <T> void validate(T worldgenElement, Registry<T> registry, String worldgenElementType, Codec<T> codec, ResourceLocation biomeName) {
+
+        // Checks to make sure the element can be turned into JSON safely or else Minecraft will explode with vague errors.
+        JsonElement worldgenElementJSON;
+        try {
+            worldgenElementJSON = codec.encode(worldgenElement, JsonOps.INSTANCE, JsonOps.INSTANCE.empty()).get().left().orElseThrow(UnsupportedOperationException::new);
```

Comment:
Is this try/catch necessary? encode won't throw, so the only thing that can is your own `#orElseThrow` so at that point just throw the exception from there. I would also consider using `#getOrThrow` on the dataResult instead to be able to output the parsing error message as well.

Reply:
I thought I had moments when it crash but you're probably right. I'll make the change but can you check out the change to make sure it looks right? I haven't used getOrThrow before